Lib/glyphsLib/filters/eraseOpenCorners.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+from fontTools.pens.basePen import BasePen
+from fontTools.misc.bezierTools import (
+    segmentSegmentIntersections,
+    _split_segment_at_t,
+)
+from ufo2ft.filters import BaseFilter
+
+logger = logging.getLogger(__name__)
+
+
+def _pointIsLeftOfLine(line, aPoint):
+    a, b = line
+    return (
+        (b[0] - a[0]) * (aPoint[1] - a[1]) - (b[1] - a[1]) * (aPoint[0] - a[0])
+    ) >= 0
+
+
+class EraseOpenCornersPen(BasePen):
```

Comment:
yours is not really a "drawing pen" like the ones BasePen was designed to work as base class, you're not using any of its built-in facilities. I think what you actually want to sub-class is the FilterPen, or even better the ContourFilterPen, which takes another pen as parameter and accummulates the drawing commands, then at the end of a contour it "flushes" them to be processed by its filter.

https://github.com/fonttools/fonttools/blob/2687c12c2e9dc372e3f35ee5e7318c20c1519c26/Lib/fontTools/pens/filterPen.py#L80

Reply:
oh I see, you're using `BasePen._getCurrentPoint`. ok

Reply:
I looked at ContourFilterPen - that was what you told me to use initially - but it doesn't do what we want here. We want to work on segments, not operations, so I use the pen class to accumulate the segment information and then work on it at the end.

Please don't get me started on "drawing" pens versus non-drawing "pens". It never ends well.